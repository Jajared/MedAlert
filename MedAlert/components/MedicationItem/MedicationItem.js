import { StyleSheet, Text, View, SafeAreaView, Image, TouchableOpacity, Animated } from "react-native";
import { Swipeable } from "react-native-gesture-handler";

export default function MedicationItem({ props, navigation, setAcknowledged }) {
  const medicationData = props.item;
  const pillIcon = require("../../assets/pill-icon.png");
  const syrupIcon = require("../../assets/syrup-icon.png");
  const rightActions = (progress, dragX) => {
    const scale = dragX.interpolate({
      inputRange: [-50, 0],
      outputRange: [1, 0],
      extrapolate: "clamp",
    });
    return (
      <TouchableOpacity onPress={() => setAcknowledged(medicationData.id)}>
        <View style={styles.rightAction}>
          <Animated.Text style={[styles.actionText, { transform: [{ scale }] }]}>Consumed</Animated.Text>
        </View>
      </TouchableOpacity>
    );
  };
  function getIcon() {
    const type = medicationData.Type;
    if (type === "Pill") {
      return pillIcon;
    } else if (type === "Liquid") {
      return syrupIcon;
    }
  }

  function getTime() {
    const time = medicationData.Instructions.FirstDosageTiming;
    const date = new Date();
    date.setHours(Math.floor(time / 60), time % 60);
    return date.toLocaleTimeString([], { hour: "2-digit", minute: "2-digit", hour12: true });
  }

  return (
    <SafeAreaView style={styles.container}>
      <Swipeable renderRightActions={rightActions}>
        <View style={styles.itemContainer}>
          <Text style={styles.timeSection}>{getTime()}</Text>
          <View style={styles.textContainer}>
            <Image source={getIcon()} style={styles.icon} />
            <View style={styles.medicationInfo}>
              <Text style={{ fontWeight: "bold", fontSize: 20 }}>{medicationData.Name}</Text>
              <Text>{medicationData.Purpose}</Text>
              <Text>{medicationData.Instructions.TabletsPerIntake} tablets per intake</Text>
            </View>
          </View>
        </View>
      </Swipeable>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
  },
  timeSection: {
    flex: 1,
    fontWeight: "bold",
    fontSize: 20,
    width: "90%",
    marginBottom: 10,
    marginTop: 10,
  },
  icon: {
    flex: 1,
    width: "80%",
    height: "80%",
    resizeMode: "contain",
    marginLeft: 10,
  },
  itemContainer: {
    flex: 1,
    flexDirection: "column",
    backgroundColor: "#FAF6E0",
    borderColor: "black",
    borderWidth: 1,
    borderRadius: 15,
    margin: 15,
    alignItems: "center",
    height: 100,
    width: "90%",
  },
  textContainer: {
    flex: 3,
    flexDirection: "row",
  },
  medicationInfo: {
    flex: 3,
    marginLeft: 20,
  },
  rightAction: {
    backgroundColor: "green",
    justifyContent: "center",
    height: 100,
    margin: 15,
    borderRadius: 15,
  },
  actionText: {
    color: "#fff",
    fontWeight: "600",
    fontSize: 16,
    padding: 10,
  },
});
