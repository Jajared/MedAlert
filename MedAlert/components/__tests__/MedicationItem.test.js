import { render, screen, fireEvent } from "@testing-library/react-native";
import MedicationItem from "../MedicationItem";
import { expect, jest, test } from "@jest/globals";

jest.mock("expo-font");
jest.mock("expo-asset");

const navigationObject = {
  navigate: jest.fn(),
  goBack: jest.fn(),
};

const testScheduledItem = {
  Name: "test",
  Purpose: "test",
  Type: "test",
  Instructions: {
    TabletsPerIntake: 1,
    FrequencyPerDay: 1,
    Specifications: "test",
    FirstDosageTiming: 1,
  },
  Acknowledged: false,
  id: 1,
  notificationId: "test",
};

describe("MedicationItem", () => {
  const acknowledgeReminder = jest.fn();
  const deleteReminder = jest.fn();
  // Check if renders correctly
  it("renders correctly", () => {
    const tree = render(<MedicationItem props={{ item: testScheduledItem }} setAcknowledged={acknowledgeReminder} deleteReminder={deleteReminder} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  // Check number of children
  it("has 1 child", () => {
    render(<MedicationItem props={{ item: testScheduledItem }} setAcknowledged={acknowledgeReminder} deleteReminder={deleteReminder} />);
    expect(screen.getByTestId("MedicationItem").children.length).toBe(1);
  });
  // Check if button pressed acknowledges reminder
  it("Acknowledges reminder", () => {
    render(<MedicationItem props={{ item: testScheduledItem }} setAcknowledged={acknowledgeReminder} deleteReminder={deleteReminder} />);
    fireEvent.press(screen.getByTestId("AcknowledgeButton"));
    expect(acknowledgeReminder).toBeCalled();
  });
  // Check if button pressed deletes reminder
  it("Deletes reminder", () => {
    render(<MedicationItem props={{ item: testScheduledItem }} setAcknowledged={acknowledgeReminder} deleteReminder={deleteReminder} />);
    fireEvent.press(screen.getByTestId("DeleteButton"));
    expect(deleteReminder).toBeCalled();
  });
});
