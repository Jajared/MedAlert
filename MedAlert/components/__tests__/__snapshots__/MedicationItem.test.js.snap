// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedicationItem renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
  testID="MedicationItem"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#FAF0D7",
          "borderColor": "#FAF0D7",
          "borderRadius": 15,
          "borderWidth": 1,
          "flex": 1,
          "flexDirection": "column",
          "height": 100,
          "margin": 15,
          "width": "90%",
        },
        {
          "backgroundColor": "#FF8989",
        },
      ]
    }
  >
    <Text
      style={
        {
          "flex": 1,
          "fontSize": 18,
          "fontWeight": "bold",
          "marginTop": 10,
          "width": "90%",
        }
      }
    >
      12:01 AM
    </Text>
    <View
      style={
        {
          "flex": 3,
          "flexDirection": "row",
        }
      }
    >
      <Image
        style={
          {
            "flex": 1,
            "height": "80%",
            "marginLeft": 10,
            "resizeMode": "contain",
            "width": "80%",
          }
        }
      />
      <View
        style={
          {
            "flex": 2,
            "marginLeft": 20,
          }
        }
      >
        <View
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "flex": undefined,
              },
            ]
          }
        >
          <Text
            animationType="auto"
            bounce={true}
            bounceDelay={50}
            bounceSpeed={50}
            duration={3000}
            easing={[Function]}
            isInteraction={true}
            loop={true}
            marqueeDelay={1000}
            marqueeOnMount={true}
            numberOfLines={1}
            scrollSpeed={150}
            style={
              [
                {
                  "fontSize": 16,
                  "fontWeight": "bold",
                },
                {
                  "opacity": 0,
                },
              ]
            }
            useNativeDriver={true}
          >
            test
          </Text>
          <RCTScrollView
            display="flex"
            horizontal={true}
            onContentSizeChange={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            scrollEnabled={false}
            scrollEventThrottle={16}
            showsHorizontalScrollIndicator={false}
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                false,
              ]
            }
          >
            <View>
              <Text
                animationType="auto"
                bounce={true}
                bounceDelay={50}
                bounceSpeed={50}
                collapsable={false}
                duration={3000}
                easing={[Function]}
                forwardedRef={[Function]}
                isInteraction={true}
                loop={true}
                marqueeDelay={1000}
                marqueeOnMount={true}
                numberOfLines={1}
                scrollSpeed={150}
                style={
                  {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "transform": [
                      {
                        "translateX": 0,
                      },
                    ],
                    "width": null,
                  }
                }
                useNativeDriver={true}
              >
                test
              </Text>
            </View>
          </RCTScrollView>
        </View>
        <View
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "flex": undefined,
              },
            ]
          }
        >
          <Text
            animationType="auto"
            bounce={true}
            bounceDelay={50}
            bounceSpeed={50}
            duration={3000}
            easing={[Function]}
            isInteraction={true}
            loop={true}
            marqueeDelay={1000}
            marqueeOnMount={true}
            numberOfLines={1}
            scrollSpeed={150}
            style={
              [
                {},
                {
                  "opacity": 0,
                },
              ]
            }
            useNativeDriver={true}
          >
            test
          </Text>
          <RCTScrollView
            display="flex"
            horizontal={true}
            onContentSizeChange={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            scrollEnabled={false}
            scrollEventThrottle={16}
            showsHorizontalScrollIndicator={false}
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                false,
              ]
            }
          >
            <View>
              <Text
                animationType="auto"
                bounce={true}
                bounceDelay={50}
                bounceSpeed={50}
                collapsable={false}
                duration={3000}
                easing={[Function]}
                forwardedRef={[Function]}
                isInteraction={true}
                loop={true}
                marqueeDelay={1000}
                marqueeOnMount={true}
                numberOfLines={1}
                scrollSpeed={150}
                style={
                  {
                    "transform": [
                      {
                        "translateX": 0,
                      },
                    ],
                    "width": null,
                  }
                }
                useNativeDriver={true}
              >
                test
              </Text>
            </View>
          </RCTScrollView>
        </View>
        <View
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "flex": undefined,
              },
            ]
          }
        >
          <Text
            animationType="auto"
            bounce={true}
            bounceDelay={50}
            bounceSpeed={50}
            duration={3000}
            easing={[Function]}
            isInteraction={true}
            loop={true}
            marqueeDelay={1000}
            marqueeOnMount={true}
            numberOfLines={1}
            scrollSpeed={150}
            style={
              [
                {},
                {
                  "opacity": 0,
                },
              ]
            }
            useNativeDriver={true}
          >
            1
             
            ml per Intake
          </Text>
          <RCTScrollView
            display="flex"
            horizontal={true}
            onContentSizeChange={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            scrollEnabled={false}
            scrollEventThrottle={16}
            showsHorizontalScrollIndicator={false}
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                false,
              ]
            }
          >
            <View>
              <Text
                animationType="auto"
                bounce={true}
                bounceDelay={50}
                bounceSpeed={50}
                collapsable={false}
                duration={3000}
                easing={[Function]}
                forwardedRef={[Function]}
                isInteraction={true}
                loop={true}
                marqueeDelay={1000}
                marqueeOnMount={true}
                numberOfLines={1}
                scrollSpeed={150}
                style={
                  {
                    "transform": [
                      {
                        "translateX": 0,
                      },
                    ],
                    "width": null,
                  }
                }
                useNativeDriver={true}
              >
                1
                 
                ml per Intake
              </Text>
            </View>
          </RCTScrollView>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 2,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="AcknowledgeButton"
        >
          <Image
            source={1}
            style={
              {
                "height": 25,
                "marginHorizontal": 15,
                "width": 25,
              }
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="DeleteButton"
        >
          <Image
            source={1}
            style={
              {
                "height": 25,
                "marginHorizontal": 15,
                "width": 25,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
